dref/views.py

```diff
@@ -0,0 +1,152 @@
+                {
+                    'key': disaster.value,
+                    'value': disaster.label
+                } for disaster in Dref.DisasterCategory
+            ],
+            'planned_interventions': [
+                {
+                    'key': intervention[0],
+                    'value': intervention[1]
+                } for intervention in PlannedIntervention.Title.choices
+            ],
+            'needs_identified': [
+                {
+                    'key': need[0],
+                    'value': need[1]
+                } for need in IdentifiedNeed.Title.choices
+            ],
+            'national_society_actions': [
+                {
+                    'key': action[0],
+                    'value': action[1]
+                } for action in NationalSocietyAction.Title.choices
+            ],
+            'users': [
+                {
+                    'id': user.id,
+                    'email': user.email,
+                    'username': user.username,
+                    'first_name': user.first_name,
+                    'last_name': user.last_name
+                } for user in User.objects.all()
+            ]
+        }
+        return response.Response(options)
+
+
+class NationalSocietyActionViewSet(viewsets.ModelViewSet):
+    serializer_class = NationalSocietyActionSerializer
+    permission_classes = [permissions.IsAuthenticated]
+    queryset = NationalSocietyAction.objects.all()
+
+
+class PlannedInterventionViewSet(viewsets.ModelViewSet):
+    serializer_class = PlannedInterventionSerializer
+    permission_classes = [permissions.IsAuthenticated]
+    queryset = PlannedIntervention.objects.all()
+
+
+class IdentifiedNeedViewSet(viewsets.ModelViewSet):
+    serializer_class = IdentifiedNeedSerializer
+    permission_class = [permissions.IsAuthenticated]
+    queryset = IdentifiedNeed.objects.all()
+
+
+class DrefFileViewSet(
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    viewsets.GenericViewSet
+):
+    permission_class = [permissions.IsAuthenticated]
+    serializer_class = DrefFileSerializer
+
+    def get_queryset(self):
+        return DrefFile.objects.filter(created_by=self.request.user)
+
+    @action(
+        detail=False,
+        url_path='multiple',
+        methods=['POST'],
+        permission_classes=[permissions.IsAuthenticated],
+    )
+    def multiple_file(self, request, pk=None, version=None):
+        # converts querydict to original dict
+        files = dict((request.data).lists())['file']
+        data = [{'file': file} for file in files]
+        file_serializer = DrefFileSerializer(data=data, context={'request': request}, many=True)
+        if file_serializer.is_valid():
+            file_serializer.save()
+            return response.Response(file_serializer.data, status=status.HTTP_201_CREATED)
+        else:
+            return response.Response(file_serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Let's save after all are validated. If we don't do that then files will be uploaded but the DrefFile will be rolled back creating orphan files in the server.

Reply:
Mistake review.